_posts/2021-03-09-bezier-fitting.md

```diff
@@ -0,0 +1,196 @@
+---
+layout: post
+title:  "Fitting cubic Bézier curves"
+date:   2021-03-09 16:10:42 -0700
+categories: [curves]
+---
+<!-- I should figure out a cleaner way to do this include, rather than cutting and pasting. Ah well.-->
+<script type="text/x-mathjax-config">
+	MathJax.Hub.Config({
+		tex2jax: {
+			inlineMath: [['$', '$']]
+		}
+	});
+</script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
+
+Cubic beziers are by far the most common curve representation, used both for design and rendering. A fundamental problem is *curve fitting,* or determining the cubic bezier that's closest to some source curve. Applications include simplifying existing paths, efficiently representing the [parallel curve], and rendering other spline representations such as Euler spiral or [hyperbezier]. A specific feature where curve fitting will be used in [Runebender] is deleting a smooth on-curve point. The intent is to merge two beziers into one, which is another way of saying to find a bezier which most closely approximates the curve formed of the two existing bezier segments.
+
+In spite of the importance of the curve-fitting problem and literature going back more than 30 years, there has been until now no fully satisfactory solution. Existing approaches either miss the optimum solution, are very slow (unsuitable for interactive use), or both.
+
+The main reason the problem is so hard is a basic but little-known fact about cubic beziers: C-shaped beziers come in triplets of very similar shape but quite different parameters. As a result, in general when fitting some arbitrary source curve, there are three local minima. Approaches based on iterative approximation (in addition to being slow) are likely to find just one, potentially missing a closer fit of one of the others.
```

Comment:
```suggestion
The main reason the problem is so hard is a fundamental but little-known property of cubic beziers: C-shaped beziers come in triplets of very similar shape but quite different parameters. As a result, when fitting some arbitrary source curve, there are commonly three local minima. Approaches based on iterative approximation (in addition to being slow) are likely to find just one, potentially missing a closer fit of one of the others.
```

- define C-shaped, I think
- what's a "triplet"? I would say something like, "There tend to be three sets of parameters that produce very similar shapes"; maybe "extremely" or something more emphatic in place of "very".